### PR TITLE
update bindings.json settings for sql extension

### PR DESCRIPTION
**Pull request by MaddyDev**
Added the functions.json settings for non dotnet languages for Sql extension. (Below is the list of attributes added)
{
      "name": "products",
      "type": "sql",
      "direction": "in/out",
      "commandText": "select * from Products where Cost = @Cost",
      "commandType": "Text",
      "parameters": "@Cost={cost}",
      "connectionStringSetting": "SqlConnectionString"
   }

### CODE REVIEW

File: Functions.Templates/Resources/Resources.resx [hunk 1]
```diff
@@ -1945,4 +1945,52 @@ function app.</value>
...
+    <value>Azure Sql Input Binding</value>
+  </data>
+  <data name="sqlIn_name_label" xml:space="preserve">
+    <value>Name</value>
+  </data>
+  <data name="sqlIn_name_help" xml:space="preserve">
+    <value>The name used to identify this binding in your code.</value>
+  </data>
+  <data name="sqlIn_commandText_label" xml:space="preserve">
+    <value>Sql Command Text</value>
```
Comment (Charles-Gagnon) [inline Functions.Templates/Resources/Resources.resx:1958]:
```suggestion
    <value>SQL Command Text</value>

```

File: Functions.Templates/Resources/Resources.resx [hunk 2]
```diff
@@ -1945,4 +1945,52 @@ function app.</value>
   <data name="authenticationEvent_trigger_IssuanceStartVer_help" xml:space="preserve">
     <value>API Schema Version for OnTokenIssuanceStart Event</value>
   </data>
+  <data name="sqlIn_displayName" xml:space="preserve">
+    <value>Azure Sql Input Binding</value>
```
Comment (Charles-Gagnon) [inline Functions.Templates/Resources/Resources.resx:1949]:
```suggestion
    <value>Azure SQL Input Binding</value>

```

File: Functions.Templates/Resources/Resources.resx [hunk 3]
```diff
@@ -1945,4 +1945,52 @@ function app.</value>
...
+    <value>The Transact-SQL query type which is either Text or StoredProcedure executed by the binding.</value>
+  </data>
+  <data name="sqlIn_parameters_label" xml:space="preserve">
+    <value>Parameters</value>
+  </data>
+  <data name="sqlIn_parameters_help" xml:space="preserve">
+    <value>Parameter values passed to the command during execution as a single string. Must follow the format `@param1=param1,@param2=param2`.</value>
+  </data>
+  <data name="sql_connection_label" xml:space="preserve">
+    <value>Sql Connection string setting</value>
```
Comment (Charles-Gagnon) [inline Functions.Templates/Resources/Resources.resx:1976]:
```suggestion
    <value>SQL Connection string setting</value>

```

File: Functions.Templates/Resources/Resources.resx [hunk 4]
```diff
@@ -1945,4 +1945,52 @@ function app.</value>
...
+  </data>
+  <data name="sql_connection_help" xml:space="preserve">
+    <value>The name of the app setting containing your Sql binding connection string.</value>
```
Comment (Charles-Gagnon) [inline Functions.Templates/Resources/Resources.resx:1979]:
```suggestion
    <value>The name of the app setting containing your SQL binding connection string.</value>

```

File: Functions.Templates/Resources/Resources.resx [hunk 5]
```diff
@@ -1945,4 +1945,52 @@ function app.</value>
...
+  </data>
+  <data name="sqlOut_displayName" xml:space="preserve">
+    <value>Azure Sql Output Binding</value>
```
Comment (Charles-Gagnon) [inline Functions.Templates/Resources/Resources.resx:1982]:
```suggestion
    <value>Azure SQL Output Binding</value>

```

File: Functions.Templates/Resources/Resources.resx [hunk 6]
```diff
@@ -1945,4 +1945,52 @@ function app.</value>
...
+  </data>
+  <data name="sqlOut_name_label" xml:space="preserve">
+    <value>Name</value>
+  </data>
+  <data name="sqlOut_name_help" xml:space="preserve">
+    <value>The name of the variable that represents the entity in function code.</value>
+  </data>
+  <data name="sqlOut_commandText_label" xml:space="preserve">
+    <value>Sql Table Name</value>
```
Comment (Charles-Gagnon) [inline Functions.Templates/Resources/Resources.resx:1991]:
```suggestion
    <value>SQL Table Name</value>

```

File: Functions.Templates/Resources/Resources.resx (hunk 4 above)
Comment (chlafreniere) [inline Functions.Templates/Resources/Resources.resx:1979]: @dzsquared should we say "Sql binding" here or "Azure Sql Binding"

File: Functions.Templates/Resources/Resources.resx [hunk 7]
```diff
@@ -1945,4 +1945,52 @@ function app.</value>
...
+  </data>
+  <data name="sqlOut_commandText_help" xml:space="preserve">
+    <value>The name of the table being written to by the binding.</value>
```
Comment (Charles-Gagnon) [inline Functions.Templates/Resources/Resources.resx:1994]:
```suggestion
    <value>The name of the table being upserted to by the binding.</value>

```

File: Functions.Templates/Resources/Resources.resx [hunk 8]
```diff
@@ -1945,4 +1945,52 @@ function app.</value>
...
+    <value>Sql Command Text</value>
+  </data>
+  <data name="sqlIn_commandText_help" xml:space="preserve">
+    <value>The Transact-SQL query command or name of the stored procedure executed by the binding.</value>
+  </data>
+  <data name="sqlIn_commandType_label" xml:space="preserve">
+    <value>Text or StoredProcedure</value>
+  </data>
+  <data name="sqlIn_commandType_help" xml:space="preserve">
+    <value>The Transact-SQL query type which is either Text or StoredProcedure executed by the binding.</value>
```
Comment (Charles-Gagnon) [inline Functions.Templates/Resources/Resources.resx:1967]:
```suggestion
    <value>The command type used to identify what type of value commandText is, either Text for a T-SQL query or StoredProcedure for a stored procedure.</value>

```

File: Functions.Templates/Bindings/bindings-bundle-v4.json [hunk 9]
```diff
@@ -1859,6 +1859,116 @@
...
+          "name": "commandText",
+          "value": "string",
+          "defaultValue": "Select * from object",
+          "required": true,
+          "label": "$sqlIn_commandText_label",
+          "help": "$sqlIn_commandText_help"
+        },
+        {
+          "name": "commandType",
+          "value": "string",
```
Comment (Charles-Gagnon) [inline Functions.Templates/Bindings/bindings-bundle-v4.json:1897]: This should probably be an enum type

File: Functions.Templates/Bindings/bindings-bundle-v4.json [hunk 10]
```diff
@@ -1859,6 +1859,116 @@
...
+          "help": "$sqlIn_commandType_help"
+        },
+        {
+          "name": "connectionStringSetting",
+          "value": "string",
+          "defaultValue": "SqlConnectionString",
+          "required": true,
+          "label": "$sql_connection_label",
+          "help": "$sql_connection_help",
+          "placeholder": "[variables('selectConnection')]"
```
Comment (Charles-Gagnon) [inline Functions.Templates/Bindings/bindings-bundle-v4.json:1910]: What does this evaluate to?

File: Functions.Templates/Bindings/bindings-bundle-v4.json [hunk 11]
```diff
@@ -1859,6 +1859,116 @@
...
+        {
+          "name": "commandText",
+          "value": "string",
+          "defaultValue": "object",
+          "required": true,
+          "label": "$sqlOut_commandText_label",
+          "help": "$sqlOut_commandText_help"
+        },
+        {
+          "name": "commandType",
```
Comment (Charles-Gagnon) [inline Functions.Templates/Bindings/bindings-bundle-v4.json:1955]: We shouldn't need this value for output bindings

File: Functions.Templates/Resources/Resources.resx (hunk 4 above)
Comment (dzsquared) [reply Functions.Templates/Resources/Resources.resx:1979]: in descriptions and free text, let's be consistent with "Azure SQL binding"

File: Functions.Templates/Documentation/sqlIn.md [hunk 12]
```diff
@@ -0,0 +1,42 @@
+#### Settings for a sql input binding
```
Comment (dzsquared) [inline Functions.Templates/Documentation/sqlIn.md:1]: minimally capitalize SQL, preferably use Azure SQL

File: Functions.Templates/Documentation/sqlIn.md [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+- `commandText` : The Transact-SQL query command or name of the stored procedure executed by the binding.
+- `connectionStringSetting` : The name of an app setting that contains the connection string for the database against which the query or stored procedure is being executed. This value isn't the actual connection string and must instead resolve to an environment variable name.  Optional keywords in the connection string value are [available to refine SQL bindings connectivity](https://aka.ms/sqlbindings#sql-connection-string).
+- `commandType` : A [CommandType](https://learn.microsoft.com/dotnet/api/system.data.commandtype) value, which is [Text](https://learn.microsoft.com/dotnet/api/system.data.commandtype#fields) for a query and [StoredProcedure](https://learn.microsoft.com/dotnet/api/system.data.commandtype#fields) for a stored procedure.
+- `parameters` : *optional*. Zero or more parameter values passed to the command during execution as a single string. Must follow the format `@param1=param1,@param2=param2`. Neither the parameter name nor the parameter value can contain a comma (`,`) or an equals sign (`=`).
+
+#### Sql input C# code example
```
Comment (dzsquared) [inline Functions.Templates/Documentation/sqlIn.md:15]: minimally capitalize SQL, preferably use Azure SQL

File: Functions.Templates/Documentation/sqlIn.md [hunk 14]
```diff
@@ -0,0 +1,42 @@
+#### Settings for a sql input binding
+
+See [Input Binding Overview](https://github.com/Azure/azure-functions-sql-extension/blob/main/docs/BindingsOverview.md#input-binding) for general information about the Azure SQL Input binding.
```
[the remark this answers is not in the document]
Comment (MaddyDev) [reply Functions.Templates/Documentation/sqlIn.md:3]: Followed the pattern of what other extensions had in the markdown. @soninaren How does this show up?

File: Functions.Templates/Bindings/bindings-bundle-v4.json (hunk 10 above)
Comment (MaddyDev) [reply Functions.Templates/Bindings/bindings-bundle-v4.json:1910]: https://github.com/Azure/azure-functions-templates/blob/b9c124c468bed78261f195f7fe0ddb8b74458a21/Functions.Templates/Resources/Resources.resx#L255

File: Functions.Templates/Bindings/bindings-bundle-v4.json [hunk 15]
```diff
@@ -1859,6 +1859,118 @@
...
+            {
+              "expression": "^[a-zA-Z][a-zA-Z0-9]{0,127}$",
+              "errorText": "[variables('parameterName')]"
+            }
+          ]
+        },
+        {
+          "name": "commandText",
+          "value": "string",
+          "defaultValue": "Select * from object",
```
Comment (Charles-Gagnon) [inline Functions.Templates/Bindings/bindings-bundle-v4.json:1890]: I've never really found a consistent default I like, but having at least stuff that matches our samples seems like a good idea

```suggestion
          "defaultValue": "SELECT * FROM Products",

```

File: Functions.Templates/Bindings/bindings-bundle-v4.json (hunk 10 above)
Comment (Charles-Gagnon) [reply Functions.Templates/Bindings/bindings-bundle-v4.json:1920]: Huh, wonder why the syntax is different here. @soninaren - I would expect it to be `$variables_selectConnection`, I guess this is a special field for some reason?

File: Functions.Templates/Documentation/sqlIn.md [hunk 16]
```diff
@@ -0,0 +1,42 @@
+#### Settings for a Azure SQL input binding
```
Comment (Charles-Gagnon) [inline Functions.Templates/Documentation/sqlIn.md:1]: It'd be good to find out why this is set to header level 4 as well - is this nested in such a way that it shouldn't be header level 1-3?

File: Functions.Templates/Documentation/sqlIn.md [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+- `commandText` : The Transact-SQL query command or name of the stored procedure executed by the binding.
+- `connectionStringSetting` : The name of an app setting that contains the connection string for the database against which the query or stored procedure is being executed. This value isn't the actual connection string and must instead resolve to an environment variable name.  Optional keywords in the connection string value are [available to refine SQL bindings connectivity](https://aka.ms/sqlbindings#sql-connection-string).
+- `commandType` : A [CommandType](https://learn.microsoft.com/dotnet/api/system.data.commandtype) value, which is [Text](https://learn.microsoft.com/dotnet/api/system.data.commandtype#fields) for a query and [StoredProcedure](https://learn.microsoft.com/dotnet/api/system.data.commandtype#fields) for a stored procedure.
+- `parameters` : *optional*. Zero or more parameter values passed to the command during execution as a single string. Must follow the format `@param1=param1,@param2=param2`. Neither the parameter name nor the parameter value can contain a comma (`,`) or an equals sign (`=`).
+
+#### SQL input C# code example
```
Comment (Charles-Gagnon) [inline Functions.Templates/Documentation/sqlIn.md:15]: Why only C# and JS? This goes with my above comment that it seems easier to just link to the docs directly - although I don't know the exact experience customers will be going through here so maybe there's a reason for that...

File: Functions.Templates/Documentation/sqlIn.md (hunk 16 above)
Comment (Charles-Gagnon) [inline Functions.Templates/Documentation/sqlIn.md:1]:
```suggestion
#### Settings for an Azure SQL input binding
```

File: Functions.Templates/Documentation/sqlOut.md [hunk 18]
```diff
@@ -0,0 +1,75 @@
+#### Settings for a Azure SQL output binding
```
Comment (Charles-Gagnon) [inline Functions.Templates/Documentation/sqlOut.md:1]:
```suggestion
#### Settings for an Azure SQL output binding
```

File: Functions.Templates/Documentation/sqlIn.md (hunk 17 above)
Comment (MaddyDev) [reply Functions.Templates/Documentation/sqlIn.md:15]: Again here I just followed the pattern from the other extensions (specially cosmos), all of them that I checked used header level 4 and have examples for only c# and JS @soninaren Do you know why this is set to header level 4 and have only c#, JS samples?

File: Functions.Templates/Bindings/bindings-bundle-v4.json [hunk 19]
```diff
@@ -1859,6 +1859,118 @@
...
       ]
+    },
+    {
+      "type": "sql",
+      "displayName": "$sqlIn_displayName",
+      "direction": "in",
+      "enabledInTryMode": false,
+      "documentation": "$content=Documentation\\sqlIn.md",
+      "extension": {
+        "id": "Microsoft.Azure.WebJobs.Extensions.Sql", "version": "1.0.122-preview"
```
Comment (Charles-Gagnon) [inline Functions.Templates/Bindings/bindings-bundle-v4.json:1870]: These should be updated too, right?

**Comment by MaddyDev** [comment]
@soninaren merging this in since all the checks passed and our team approved offline. Please reach out to us know if you've any questions/concerns. And it would be great to have our comments answered as well.

**Pull request merged**